lib/components/Dialog/Dialog.test.tsx

```diff
@@ -161,6 +161,7 @@ describe('Dialog', () => {
     userEvent.click(dialogOpenButton);
 
     await waitFor(() => queryByRole('dialog'));
+    await new Promise((resolve) => setTimeout(resolve, 500)); // Wait for transition
 
     expect(queryAllByRole('textbox').length).toBe(0);
```

Comment:
I think this would be neater to wait for the assertion to be true:
```suggestion
    await waitFor(() => expect(queryAllByRole('textbox').length).toBe(0))
```